app/src/main/java/com/keylesspalace/tusky/di/NetworkModule.kt

```diff
@@ -125,43 +138,97 @@ object NetworkModule {
-                    )
-                }
-            }
-            .build()
+        if (BuildConfig.DEBUG) {
+            builder.addInterceptor(
+                HttpLoggingInterceptor().apply { level = HttpLoggingInterceptor.Level.BASIC }
+            )
+        }
+        return builder.build()
     }
 
     @Provides
     @Singleton
-    fun providesRetrofit(httpClient: OkHttpClient, moshi: Moshi): Retrofit {
-        return Retrofit.Builder().baseUrl("https://" + MastodonApi.PLACEHOLDER_DOMAIN)
+    fun providesRetrofit(
+        httpClient: OkHttpClient,
+        moshi: Moshi,
+        @Named("defaultPort") defaultPort: Int,
+        @Named("defaultScheme") defaultScheme: String
+    ): Retrofit {
+        return Retrofit.Builder()
+            .baseUrl("$defaultScheme${MastodonApi.PLACEHOLDER_DOMAIN}:$defaultPort")
             .client(httpClient)
             .addConverterFactory(MoshiConverterFactory.create(moshi))
             .addCallAdapterFactory(NetworkResultCallAdapterFactory.create())
             .build()
     }
 
     @Provides
-    @Singleton
-    fun providesApi(retrofit: Retrofit): MastodonApi = retrofit.create()
+    fun providesMastodonApi(
+        httpClient: OkHttpClient,
+        retrofit: Retrofit,
+        accountManager: AccountManager,
+        @Named("defaultPort") defaultPort: Int,
+        @Named("defaultScheme") defaultScheme: String
+    ): MastodonApi {
+        return provideApi(httpClient, retrofit, accountManager, defaultPort, defaultScheme)
+    }
 
     @Provides
-    @Singleton
-    fun providesMediaUploadApi(retrofit: Retrofit, okHttpClient: OkHttpClient): MediaUploadApi {
+    fun providesMediaUploadApi(
+        retrofit: Retrofit,
+        okHttpClient: OkHttpClient,
+        accountManager: AccountManager,
+        @Named("defaultPort") defaultPort: Int,
+        @Named("defaultScheme") defaultScheme: String
+    ): MediaUploadApi {
         val longTimeOutOkHttpClient = okHttpClient.newBuilder()
             .readTimeout(100, TimeUnit.SECONDS)
             .writeTimeout(100, TimeUnit.SECONDS)
             .build()
 
+        return provideApi(longTimeOutOkHttpClient, retrofit, accountManager, defaultPort, defaultScheme)
+    }
+
+    @VisibleForTesting
+    inline fun <reified T> provideApi(
+        httpClient: OkHttpClient,
+        retrofit: Retrofit,
+        accountManager: AccountManager,
+        @Named("defaultPort") defaultPort: Int,
+        @Named("defaultScheme") defaultScheme: String
+    ): T {
+        val currentAccount = accountManager.activeAccount
         return retrofit.newBuilder()
-            .client(longTimeOutOkHttpClient)
+            .apply {
+                if (currentAccount != null) {
+                    baseUrl("$defaultScheme${currentAccount.domain}:$defaultPort")
+                }
+            }
+            .callFactory { originalRequest ->
+                var request = originalRequest
```

Comment:
i think that's a lot of code for DI module tbh, it shows in test that you ask it from DI which seems bit off to me, but not critical ofc, we can refactor this later